source/common/http/http1/conn_pool.cc

```diff
@@ -203,7 +203,8 @@ void ConnPoolImpl::onResponseComplete(ActiveClient& client) {
   if (!client.stream_wrapper_->encode_complete_) {
     ENVOY_CONN_LOG(debug, "response before request complete", *client.codec_client_);
     onDownstreamReset(client);
-  } else if (client.stream_wrapper_->saw_close_header_ || client.codec_client_->remoteClosed()) {
+  } else if (client.stream_wrapper_->saw_close_header_ || client.codec_client_->remoteClosed() ||
+             client.codec_client_->protocol() == Protocol::Http10) {
```

Comment:
http1.0 could enable keepalive. It's probably "client.codec_client_->protocol() == Protocol::Http10 && not keep alive?"

Reply:
https://www.w3.org/Protocols/HTTP/1.0/draft-ietf-http-spec.html

There is no keep-alive for 1.0

Reply:
https://en.wikipedia.org/wiki/HTTP_persistent_connection

Sorry, that was the origional... Yes, have to check that.

Reply:
Done.